### ISSUE TITLE
Emscripten build fails on windows/MinGW (but works ok on WSL)

### ISSUE THREAD

**Issue by UNDEFINED-BEHAVIOR** [rokups/rbfx#91]
As the title said.

The parent project documentation already warned that MinGW build might not work but it currently does. However doesn't on this project. 

Since it's not on the CI tests if it's not meant to be supported maybe a few warning will prevent future adopter from wasting a few days to find out that it doesn't works..

(btw latest SDL + emscripten upgrade experiment appears to be working here (on my machine :^) ))

**Comment by rokups** [comment]
I am confused. Which one fails, mingw or emscripten? Also provide errors 
you are getting. At the moment emscripten is only tested on Linux and mingw 
32bit builds are broken, but 64bit builds were recently fixed so should 
work. Thanks for taking interest in the project ;)

**Comment by UNDEFINED-BEHAVIOR** [comment]
Sorry, I didn't save it. I might try recreate it later.

A bit of both:
compiling via cmd with emsdk and it's mingw64bit env activated: EMSCRIPTEN/__EMSCRIPTEN __ define mysteriously disappears on the second build/ second config calls and causes havoc in places.
Standalone mingw64 and msys2: Can't even activate emsdk because python script raises
https://github.com/emscripten-core/emsdk/blob/169a3ac27c08b69169a7e1e57a37407da439e273/emsdk#L1642

I guess nothing have to be done since it's working under WSL, just to let you know.
I think the most hassle free way to build on windows might be to either use WSL or a VM matching test builds.

**Comment by rokups** [comment]
Alright ill try myself. Emscripten is still very much experimental and i did not put much time into it. CI cant even build entire thing properly because long linking times are longer than what travis allows for one build job to run.

**Comment by rokups** [comment]
I tried myself and bumped into same issues. It seems to me that there are few problems at play. Emscripten SDK does not appear to support mingw.

I am able to generate build directory using visual studio, but since it is visual studio it wont use emscripten compiler so that is a no-go as well. Using various other generators failed.

It almost worked with Ninja generator: `cmake -G Ninja -DCMAKE_BUILD_WITH_INSTALL_RPATH=ON -DBUILD_SHARED_LIBS=OFF -DURHO3D_SAMPLES=ON -DEMSCRIPTEN=1 -DCMAKE_TOOLCHAIN_FILE=../CMake/Toolchains/Emscripten.cmake -DEMSCRIPTEN_ROOT_PATH=C:\Urho3D\emsdk-master\fastcomp\emscripten ..`. Emscripten insisted on running some python2 script which was not present. I fixed that with `mklink c:\windows\python2.exe c:\Python27\python.exe`. And now it fails with "filename too long" error. It may be possible to compile web build on native windows if you absolutely need to, but for now it is unsupported configuration. Ill make a note in readme.
